src/lib/backward-search/backward-search.ts

```diff
@@ -287,7 +287,7 @@ export function createGoalsGraph<LU extends LearningUnit>(
         // Create a node in the graph for each required skills
         // Create a edge from the learning unit (source) to each required skills (sink)
         // Add the new nodes (required skills) to the nodes list for analyzing
-        lu.requiredSkills.forEach(req => {
+        lu.requiredSkills.extractSkills().forEach(req => {
```

Comment:
Should we consider the without here for potential speed up? However, it's more important not to miss any important skills.

Reply:
"Without" will remove potential skills from the search space, if there are parent/children skill relationship.